shell/mixins/harvester-vm/index.js

```diff
@@ -863,24 +847,63 @@ export default {
       this.networkScript = value;
     },
 
+    mergeSSHAuthorizedKeys(yaml) {
+      try {
+        const userDataDoc = YAML.parseDocument(yaml);
+        const sshAuthorizedKeysSeq = userDataDoc.get('ssh_authorized_keys');
+        let sshAuthorizedKeys = sshAuthorizedKeysSeq?.toJSON() || [];
+
+        if (sshAuthorizedKeys.length > 0) {
+          const sshList = new Set([...this.getSSHListValue(this.sshKey), ...sshAuthorizedKeys]);
+
+          sshAuthorizedKeys = [...sshList];
+        } else {
+          sshAuthorizedKeys = this.getSSHListValue(this.sshKey);
+        }
+
+        return sshAuthorizedKeys;
+      } catch (e) {
+        return [];
+      }
+    },
+
+    deleteYamlDocProp(doc, prop) {
+      try {
+        const item = doc.getIn([])?.items[0];
+        const key = item?.key;
+        const commentBefore = key?.commentBefore;
+
+        if (key && commentBefore?.includes('cloud-config') && key.source === prop[prop.length - 1]) {
```

Comment:
it would be nice to wrap in readable variables `commentBefore?.includes('cloud-config')` and `key.source === prop[prop.length - 1]`

Reply:
Done